src/gt4py/cartesian/frontend/defir_to_gtir.py

```diff
@@ -82,9 +84,12 @@ def _make_literal(v: numbers.Number) -> gtir.Literal:
             dtype = common.DataType.INT64
         elif isinstance(v, float):
             dtype = common.DataType.FLOAT64
+        elif isinstance(v, np.float32):
+            dtype = common.DataType.FLOAT32
         else:
             print(
-                f"Warning: Only INT64 and FLOAT64 literals are supported currently. Implicitly upcasting `{v}` to FLOAT64"
+                "Warning: Only INT64, FLOAT64 and FLOAT32 (via np.float32) literals are "
+                f"supported currently. Implicitly upcasting `{v}` of type {type(v)} to FLOAT64"
```

Comment:
Why not support all the integer types as well?

Reply:
Actually this was a quiet upcast, which is not great. So I added ints and turned unknown type into a TypeError, wadya think?